examples/asr/emformer_rnnt/pipeline_demo.py

```diff
@@ -1,38 +1,39 @@
+#!/usr/bin/env python
+"""Train the SentencePiece model by using the transcripts of MuST-C release v2.0 training set.
+Example:
+python pipeline_demo.py --model-type librispeech --dataset-path ./datasets/librispeech
+"""
 import logging
 import pathlib
-from argparse import ArgumentParser
+from argparse import ArgumentParser, RawTextHelpFormatter
+from functools import partial
 
 import torch
 import torchaudio
-from common import MODEL_TYPE_LIBRISPEECH, MODEL_TYPE_TEDLIUM3
+from common import MODEL_TYPE_LIBRISPEECH, MODEL_TYPE_MUSTC, MODEL_TYPE_TEDLIUM3
+from mustc.dataset import MUSTC
 from torchaudio.pipelines import EMFORMER_RNNT_BASE_LIBRISPEECH
-from torchaudio.prototype.pipelines import EMFORMER_RNNT_BASE_TEDLIUM3
+from torchaudio.prototype.pipelines import EMFORMER_RNNT_BASE_MUSTC, EMFORMER_RNNT_BASE_TEDLIUM3
 
+logger = logging.getLogger(__name__)
 
-logger = logging.getLogger()
 
+BUNBLES = {
+    MODEL_TYPE_LIBRISPEECH: EMFORMER_RNNT_BASE_LIBRISPEECH,
+    MODEL_TYPE_MUSTC: EMFORMER_RNNT_BASE_MUSTC,
+    MODEL_TYPE_TEDLIUM3: EMFORMER_RNNT_BASE_TEDLIUM3,
+}
 
-def get_dataset(model_type, dataset_path):
-    if model_type == MODEL_TYPE_LIBRISPEECH:
-        return torchaudio.datasets.LIBRISPEECH(dataset_path, url="test-clean")
-    elif model_type == MODEL_TYPE_TEDLIUM3:
-        return torchaudio.datasets.TEDLIUM(dataset_path, release="release3", subset="test")
-    else:
-        raise ValueError(f"Encountered unsupported model type {model_type}.")
-
-
-def get_pipeline_bundle(model_type):
-    if model_type == MODEL_TYPE_LIBRISPEECH:
-        return EMFORMER_RNNT_BASE_LIBRISPEECH
-    elif model_type == MODEL_TYPE_TEDLIUM3:
-        return EMFORMER_RNNT_BASE_TEDLIUM3
-    else:
-        raise ValueError(f"Encountered unsupported model type {model_type}.")
+DATASETS = {
+    MODEL_TYPE_LIBRISPEECH: partial(torchaudio.datasets.LIBRISPEECH, url="test-clean"),
+    MODEL_TYPE_MUSTC: partial(MUSTC, subset="tst-HE"),
+    MODEL_TYPE_TEDLIUM3: partial(torchaudio.datasets.TEDLIUM, release="release3", subset="test"),
+}
```

Comment:
Since keys of BUNDLES and DATASETS have to be same, I'd create a super dictionary.

```python
@dataclass
class Config:
    dataset: ...
    bundle: ...


_CONFIGS = {
    MODEL_TYPE_LIBRISPEECH: Config(...),
    MODEL_TYPE_MUSTC: Config(...),
    MODEL_TYPE_TEDLIUM3: Config(...),
}
```

Reply:
The dataset required `dataset_path` as the argument, here I used `partial` to accept it. How to make it in the class?

Reply:
addressed in https://github.com/pytorch/audio/pull/2222/commits/f3d19ac9ee661750a074d1e406a5f8609dddf796